packages/github-actions/teams-toolkit-cicd/src/main.ts

```diff
@@ -1,19 +1,76 @@
 import * as core from '@actions/core'
-import {wait} from './wait'
+import {ActionInputs} from './constant'
+import {Operations} from './operations'
+import {OperationType} from './enums/operationTypes'
+import {WordsToList} from './utils/words-to-list'
+import {InputsError} from './errors'
+import {Capability} from './enums/capabilities'
+import {BaseError} from './base-error'
 
 async function run(): Promise<void> {
   try {
-    const ms: string = core.getInput('milliseconds')
-    core.debug(`Waiting ${ms} milliseconds ...`) // debug is only output if you set the secret `ACTIONS_RUNNER_DEBUG` to true
+    let projectRoot = core.getInput(ActionInputs.ProjectRoot)
+    if (!projectRoot && process.env.GITHUB_WORKSPACE) {
+      projectRoot = process.env.GITHUB_WORKSPACE
+    }
+    const operationType = core.getInput(ActionInputs.OperationType)
 
-    core.debug(new Date().toTimeString())
-    await wait(parseInt(ms, 10))
-    core.debug(new Date().toTimeString())
+    if (
+      !projectRoot ||
+      !operationType ||
+      !Object.values<string>(OperationType).includes(operationType)
+    ) {
+      throw new InputsError(
+        `${ActionInputs.ProjectRoot}: ${projectRoot}, ${ActionInputs.OperationType}: ${operationType}`
+      )
+    }
 
-    core.setOutput('time', new Date().toTimeString())
+    switch (operationType) {
+      case OperationType.BuildTeamsApp: {
+        let capabilities = core.getInput(ActionInputs.Capabilities)
```

Comment:
NIT:
```
const capabilities = core.getInput(ActionInputs.Capabilities) ?? Object.values<string>(Capability).join(',')
```

Reply:
Try this later.

Reply:
It's the same reason with above. ?? doesn't work for empty string.

Reply:
Or, you can use `const capabilities = core.getInput(ActionInputs.Capabilities) || Object.values<string>(Capability).join(',')
`